controllers/azuremachine_reconciler.go

```diff
@@ -346,6 +346,7 @@ func (s *azureMachineService) createVirtualMachine(nicName string) (*infrav1.VM,
 func GetControlPlaneMachines(machineList *clusterv1.MachineList) []*clusterv1.Machine {
 	var cpm []*clusterv1.Machine
 	for _, m := range machineList.Items {
+		m := m
```

Comment:
what is this for? is it related to the change or is it an unrelated fix sneaking into the PR?

Reply:
It's related since golangci-lint won't pass unless it's fixed. Outside of that, I would have broken into another commit.

Reply:
The for loop is passing a reference to a var. If that var is not captured in the closure of the for loop, then bad things can happen.

Reply:
```
controllers/azuremachine_reconciler.go:349:33: G601: Implicit memory aliasing in for loop. (gosec)
		if util.IsControlPlaneMachine(&m) {
		                              ^
```